twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillController.java

```diff
@@ -77,7 +77,7 @@
    */
   YarnTwillController(String appName, RunId runId, ZKClient zkClient,
                       final ApplicationMasterLiveNodeData amLiveNodeData, final YarnAppClient yarnAppClient) {
-    super(runId, zkClient, Collections.<LogHandler>emptyList());
+    super(appName, runId, zkClient, amLiveNodeData.getKafkaZKConnect() != null, Collections.<LogHandler>emptyList());
```

Comment:
? I don't understand, can you give me an example? The current logic is, from an existing app discovered via ZK, if the AM live node data doesn't have the KafkaZKConnect string, that means the app was launched with log collection disabled, hence the logic here.